[Sponsored by ARK] Encryption (#24489)
PX4/PX4-Autopilot: Tools/log_encryption/README.md

# PX4 Log Encryption Tools

   Tools for generating encryption keys, building PX4 firmware with encrypted logs, downloading logs, and decrypting them.

## Usage

1. **Get the board file**:

>    To be able to use these methode you need to create your own board file in your board folder. You can just copy one that is used below.

Reviewer: ```suggestion
   In order to use these tools you need to create an `encrypted_logs` target in your target board directory. For example:
```

   ```bash
   encrypted_logs.px4board
   ```

>    Then using menuconfig you should Enable the settings: Blake2s hash algorithm, Entropy pool and strong random number generator, and Use interrupts to feed timing randomness to entropy pool.

Reviewer: ```suggestion
   Using `make menuconfig` you should enable these settings: `Blake2s hash algorithm`, `entropy pool` and `strong random number generator` and select `use interrupts` to feed timing randomness to the entropy pool.
```

   Once you have generated the keys make sure you add them to the boardconfig.

   ```bash
   make <your_board_name>_encrypted_logs menuconfig
   ```

2. **Generate Keys**:
   ```bash
   cd PX4-Autopilot/Tools/log_encryption
   python3 generate_keys.py
   ```

   Make sure you have the right key in your board file
   ```CONFIG_PUBLIC_KEY1="../../../keys/public/public_key.pub"```

3. **Build Firmware**:
   ```bash
   cd PX4-Autopilot

   AND

   make <your_board_name>_encrypted_logs

   OR

   make ark_fmu-v6x_encrypted_logs

   OR

   make ark_fpv_encrypted_logs

   OR

> 

Reviewer: excessive

   make ark_pi6x_encrypted_logs

>    AND

Reviewer: seems excessive. Maybe just provide the single example 
```
make_ark_fpv_encrypted_logs
```

>    Upload the custom fw on your flight controller and record couple of logs

Reviewer: ```suggestion
   Upload the custom firmware on your flight controller and record some logs
```

   ```

4. **Download Logs**:
   ```bash
   cd PX4-Autopilot/Tools/log_encryption

   python3 download_logs.py serial:/dev/ttyACM0:57600

   OR

   python3 download_logs.py udp:0.0.0.0:14550
   ```

   Addresses might need to be adjusted

5. **Decrypt Logs**:

>    The easiest way to run this is to have your private key and encrypted logs in te following folders respectively:

Reviewer: ```suggestion
   The easiest way to run this is to have your private key and encrypted logs in the following folders respectively:
```

   ```bash
   PX4-Autopilot/keys/private
   PX4-Autopilot/logs/encrypted
   ```
   Then run:
   ```bash
   cd PX4-Autopilot/Tools/log_encryption

   AND
   # Uses default key + default folder
   python3 decrypt_logs.py

   OR
   # Uses custom key
   python3 decrypt_logs.py path/to/private_key.pem

   OR
   # Uses custom key + specific file
   python3 decrypt_logs.py path/to/private_key.pem path/to/custom_log.ulge

   OR
   # Uses custom key + specific folder
   python3 decrypt_logs.py path/to/private_key.pem path/to/custom_log_folder

   OR
   # Uses default key + specific file
   python3 decrypt_logs.py "" path/to/custom_log.ulge

   OR

>    # Uses default key + specific folder

Reviewer: excessive, can't we just show it in standard help format?

# PX4 Log Encryption Tools

Tools for generating encryption keys, building PX4 firmware with encrypted logs, downloading logs, and decrypting them.

## Prerequisites

- Python 3.x

> - pip3 install --user pycryptodome pymavlink

Reviewer 2: Can you add this to the `Tools/setup/requirements.txt` file?

Author: Added them!

# PX4 Log Encryption Tools

>    Tools for generating encryption keys, building PX4 firmware with encrypted logs, downloading logs, and decrypting them.

Reviewer 3: Perhaps

```suggestion
Tools for generating encryption keys, building PX4 firmware with encrypted logs, downloading logs, and decrypting them.

For more information see: https://docs.px4.io/main/en/dev_log/log_encryption.html
```
